lib/stdlib/src/gen_statem.erl

```diff
@@ -438,357 +512,539 @@ handle_event(_, _, State, Data) ->
 
 
 %%%==========================================================================
-%%% Interface functions.
+%%% Callback functions
 %%%==========================================================================
 
 -doc """
-Destination to use when replying through, for example, the `t:action/0`
-`{reply,From,Reply}` to a process that has called the `gen_statem` server using
-`call/2`.
-""".
+    A [`call`](`t:external_event_type/0`) event's reply destination.
+
+    Destination to use when replying through, for example,
+    the action [`{reply, From, Reply}`](`t:reply_action/0`)
+    to a process that has called the `gen_statem` server
+    using [`call/2,3`](`call/3`).
+    """.
 -type from() ::
 	{To :: pid(), Tag :: reply_tag()}. % Reply-to specifier for call
+
+%%----------------------
 -doc "A handle that associates a reply to the corresponding request.".
 -opaque reply_tag() :: gen:reply_tag().
 
+%%----------------------
 -doc """
-If the [_callback mode_](`t:callback_mode/0`) is `handle_event_function`, the
-state can be any term. After a _state change_ (`NextState =/= State`), all
-postponed events are retried.
-""".
+    State name or state term.
+
+    If the [_callback mode_](`t:callback_mode/0`) is `handle_event_function`,
+    the state can be any term. After a _state change_ (`NextState =/= State`),
+    all postponed events are retried.
```

Comment:
"retry" sounds a bit misleading, as it usually implies a previous _failure_. I'm not sure about a better word, maybe "redeliver" or "replay"? The section above about event postponing also uses the term "retry" but explains better what is going on, so maybe just link to it?

Reply:
Well, since the event has been delivered before but the state machine refused to handle it, that is kind of a failure, so now we try again if the event will be accepted.

"Redelivered" or "replayed" also works, but for now I think "retried" is pretty ok.

Reply:
Yeah, `retried` is probably enough and understood for all common real-world use cases.

On the nitty-picky side of things, postponing does not necessarily imply a refusal or failure to _handle_ an event, but rather not _consuming_ it. A `gen_statem` may do something meaningful with an event and _still_ postpone it afterwards (like, eat your cake and still have it :cake::yum:). This is probably an unlikely and questionable edge case, so as far as I'm concerned, "retried" is fine :wink:

Reply:
That is perfectly true.

From the `gen_statem` engine's point of view, though, "retry" works better, because its objective is to have callback module consume the event, and it tries as many times it takes.